packages/roosterjs-content-model-plugins/test/tableEdit/tableInserterTest.ts

```diff
@@ -1,124 +1,108 @@
+        !editor.isDisposed && editor.dispose();
+        tInserter && tInserter.dispose();
+        const div = document.getElementById(id);
+        div?.parentNode?.removeChild(div);
+        node.parentElement?.removeChild(node);
     });
 
-    function isClickInsideInserter(click: Position, rect: DOMRect) {
-        return (
-            click.x >= rect.left &&
-            click.x <= rect.right &&
-            click.y >= rect.top &&
-            click.y <= rect.bottom
-        );
-    }
+    function runInserterTest(
+        table: ContentModelTable,
+        inserterType: string,
+        colIndex: number,
+        rowIndex: number
+    ) {
+        //Arrange
+        initialize(table);
+        const nodeHeight = 1000;
+        const nodeWidth = 1000;
+        node.style.height = `${nodeHeight}px`;
+        node.style.width = `${nodeWidth}px`;
+        node.style.overflowX = 'auto';
+        node.scrollTop = 0;
+        const target = document.getElementById(targetId);
+        editor.focus();
+
+        if (!target) {
+            fail('Table not found');
+            return;
+        }
 
-    function runInserterTest(inserterType: string, mouseEnd: Position) {
-        handler.mousemove(
-            new MouseEvent('mousemove', {
-                clientX: mouseEnd.x,
-                clientY: mouseEnd.y,
-            })
+        const targetTd = (target as HTMLTableElement).rows[rowIndex].cells[colIndex];
+        const div = document.createElement('div');
+        const onInsertSpy = jasmine.createSpy('onInsert');
+        const beforeTable = getCurrentTable(editor);
+        const rows = getTableRows(beforeTable);
+        const cols = getTableColumns(beforeTable);
+        tInserter = new TableInsertHandler(
+            div,
+            targetTd,
+            target as HTMLTableElement,
+            inserterType == HORIZONTAL_INSERTER_ID,
+            editor,
+            onInsertSpy
         );
 
-        const inserter = editor.getDocument().getElementById(inserterType);
-        if (!!inserter) {
-            const inserterRect = inserter.getBoundingClientRect();
-            if (!isClickInsideInserter(mouseEnd, inserterRect)) {
-                // Inserter is visible, but pointer is not over it
-                return 'not clickable';
-            }
-            let table = getCurrentTable(editor);
-            const rows = getTableRows(table);
-            const cols = getTableColumns(table);
-            inserter.dispatchEvent(new MouseEvent('click'));
-
-            table = getCurrentTable(editor);
-            const newRows = getTableRows(table);
-            const newCols = getTableColumns(table);
-            expect(newRows).toBe(inserterType == VERTICAL_INSERTER_ID ? rows : rows + 1);
-            expect(newCols).toBe(inserterType == HORIZONTAL_INSERTER_ID ? cols : cols + 1);
-        }
-        return !!inserter ? 'found' : 'not found';
+        //Act
+        div.click();
```

Comment:
nit: Not something we must fix but want to bring up. We can also consider not use something like "element.click()" but instead we directly call the click handler function so we don't need to rely on the browser's event routing.